client/shipping-label/views/steps/rates/list.js

```diff
@@ -1,61 +1,80 @@
 import React, { PropTypes } from 'react';
 import Dropdown from 'components/dropdown';
+import Notice from 'components/notice';
 import { translate as __ } from 'lib/mixins/i18n';
+import { sprintf } from 'sprintf-js';
+import get from 'lodash/get';
+
+const renderRateNotice = ( show ) => {
+	if ( show ) {
+		return (
+			<Notice
+				className="rate-not-available"
+				icon="info-outline"
+				showDismiss={ false }
+				text={ __( 'The service and rate chosen by the customer at checkout is not available. Please choose another.' ) }
+			/>
+		);
+	}
+};
 
 const ShippingRates = ( {
 		id,
 		selectedRates,
 		availableRates,
 		packages,
 		updateRate,
-		dimensionUnit,
-		weightUnit,
 		currencySymbol,
 		errors,
+		showRateNotice,
 	} ) => {
-	const renderTitle = ( pckg ) => {
-		return `${pckg.length} ${dimensionUnit} x ${pckg.width} ${dimensionUnit} x ${pckg.height} ${dimensionUnit} (${pckg.weight} ${weightUnit})`;
+	const renderTitle = ( pckg, idx ) => {
+		if ( 1 === packages.length ) {
+			return __( 'Choose rate' );
+		}
+		return sprintf( __( 'Choose rate: Package %(index)d' ), { index: idx + 1 } );
 	};
 
 	const renderSinglePackage = ( pckg, index ) => {
-		const selectedRate = selectedRates[ index ] || '';
-
+		const selectedRate = selectedRates[ pckg.id ] || '';
+		const packageRates = get( availableRates, [ pckg.id, 'rates' ], [] );
 		const valuesMap = { '': __( 'Select one...' ) };
-		Object.keys( availableRates[ index ] ).forEach( ( serviceId ) => {
-			const rateObject = availableRates[ index ][ serviceId ];
-			valuesMap[ serviceId ] = rateObject.name + ' (' + currencySymbol + rateObject.rate.toFixed( 2 ) + ')';
+
+		packageRates.forEach( ( rateObject ) => {
+			valuesMap[ rateObject.service_id ] = rateObject.title + ' (' + currencySymbol + rateObject.rate.toFixed( 2 ) + ')';
```

Comment:
`toFixed` means that the float `9.1` will be displayed as `"9.10"`, which is what we want for displaying money amounts.